waspc/data/Generator/templates/react-app/vite.config.ts

```diff
@@ -29,6 +29,12 @@ const defaultViteConfig = {
   build: {
     outDir: "build",
   },
+  resolve: {
+    // These packages rely on a single instance per page. Not dedpuing them
+    // causes runtime errors (e.g., hook rule violation in react, QueryClient
+    // instance error in react-query, Invariant Error in react-router-dom).
+    dedupe: ["react", "react-dom", "@tanstack/react-query", "react-router-dom"]
```

Comment:
As I commented in the DMs, I think we should pick the plugin that's the most used/supported. Then we should list these dedups explicitly like you just did.

Reply:
OK, leaving this one then (it's much more popular).